modules/tinymce/src/core/main/ts/newline/NewLineUtils.ts

```diff
@@ -84,21 +84,21 @@ const moveToCaretPosition = (editor: Editor, root: Node): void => {
   ScrollIntoView.scrollRangeIntoView(editor, rng);
 };
 
-const getEditableRoot = (dom: DOMUtils, node: Node): HTMLElement => {
+const getEditableRoot = (dom: DOMUtils, node: Node): Optional<HTMLElement> => {
   const root = dom.getRoot();
   let editableRoot: HTMLElement | undefined;
-
   // Get all parents until we hit a non editable parent or the root
   let parent: Node | null = node;
-  while (parent !== root && parent && dom.getContentEditable(parent) !== 'false') {
-    if (dom.getContentEditable(parent) === 'true') {
+  while (parent !== root && parent) {
+    if (dom.getContentEditable(parent) === 'false') {
+      return Optional.none();
+    } else if (dom.getContentEditable(parent) === 'true') {
       editableRoot = parent as HTMLElement;
     }
 
     parent = parent.parentNode;
   }
-
-  return parent !== root && editableRoot ? editableRoot : root;
+  return Optional.some(parent !== root && editableRoot ? editableRoot : root);
 };
```

Comment:
@TheSpyder Where would be a good place to move this in the core api so that it can be referenced in Lists and Advlist?

Reply:
Moved to DOMUtils